proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,471 @@
+This project board will track GitHub issues from the contribution-ready stage
+through completion by merging a pull request. New and existing GitHub labels and GitHub issue
+events will be used to drive the movement of the issue between the lanes of the
+GitHub project, which map to the stages of the contribution process.
+
+The first lane in the GitHub project contains the list of contribution-ready
+issues. This is the entry point for a user interested in making a contribution.
+
+When a contributor opens a pull request for an issue in the contribution ready
+list, it will be moved to the Implementation lane. This indicates that the
+contibution is ready for an initial review by the maintainers. The maintainers
+will perform an initial review within [TBD] days.
+
+Once the initial review is confirmed to be complete, the issue can be labeled as
+ready for review, and it moves to the 'Ready for Review' lane. These issues
+will be prioritized, selected, and labeled for review on a [TBD] basis.
+
+Once the maintainers have selected a pull request for review, a label marking it
+as ready for review will be added, and the pull request will move to the 'Under
+Review' lane of the GitHub project. This lane will provide visibility to the
+community for what the maintainers are currently working on. 
+
+After the review process is completed and a pull request is merged, the issue
+will be closed and moved to the 'Done' lane. This will provide the community
+with a list of successful contributions. 
+
+## Managing the existing backlog
+
+As of 2021-12-15, there are 188 pull requests and 436 issues in the AWS CLI
+GitHub repository. 43 pull requests and 87 issues have not been updated in more
+than a year. 95 issues have ten or more upvotes. [TBD] of the open pull requests
+are directly related to an existing feature request. [TBD] of open pull requests
+have extensive merge conflicts.
+
+Based on these numbers, we propose to review and close existing pull requests in
+two phases:
+
+1. We will use existing automated infrastructure to comment on ancient pull
+   requests, defined as no activity in more than a year and less than 10
+   upvotes. This will post a comment indicating that the issue will be closed
+   without further interaction from the community.
+2. Next, we will determine if any remaining remaining open pull requests meet
+   the criteria for "ready for contribution" or "ready for review". If a pull
+   request does not meet the criteria for either of these states, we should post
+   a comment to indicate what needs to be done and give the contributor [TBD]
+   days before the pull request is closed. If there is no response or they
+   respond that they are no longer interested in working on it, we will close
+   the pull request. If the request meets the criteria for 'ready for
+   contribution' but not 'ready for review', we will comment in the linked issue
+   that it is no longer being worked on and is available for contribution.
+
```

Comment:
We should probably also add a final step where we go through all of the open GitHub issues that do not have pull requests and add comments on what the next steps would be to be contribution ready if feasible.